app/controllers/megaliths_controller.rb

```diff
@@ -5,6 +5,11 @@ def index
 
   def show
     @megalith = Megalith.find(params[:id])
+    @trip_ids = @megalith.trip_ids
+    @all_trips = @trip_ids.map do |id|
+      Trip.find(id)
+    end
```

Comment:
if we have
has_many :trips, through: :trip_megaliths
in the megalith model, you should just be able to call
@trips = @megalith.trips
is it not?